doc/examples/example_symm.md

```diff
@@ -0,0 +1,206 @@
+}
+catch (const MoCOCrWException& e) {
+    std::cerr << "Decryption failed. It is possible that the message was modified after"
+    " encryption. " << e.what() << std::endl;
+}
+
+// Ensure that decrypted text matches the original message
+assert(std::equal(plaintext.begin(), plaintext.end(), decryptedtext.begin()));
+
+```
+
+## Steaming-mode
+
+Symmetric encryption interface also supports streaming use-case. It allows encryption of big chunks of data without loading them into memory. The implementation is optimized to avoid unnecessary copies of the input and output buffers when used with fixed block sizes in streaming en-/decryption.
+
+The following example demonstrates how to use stream mode with authenticated cipher.
+
+```c++
+
+std::string message;
+for (size_t i = 0; i < 1024; i++) {
+    message += "a string which will be concatenated multiple times-";
+}
+std::vector<uint8_t> plaintext{message.begin(), message.end()};
+
+auto secretKey = utility::cryptoRandomBytes(256/8);
+
+//
+// Encryption
+//
+
+auto encryptor = AESCipherBuilder{SymmetricCipherMode::GCM, SymmetricCipherKeySize::S_256, secretKey}.buildAuthenticatedEncryptor();
+
+std::vector<uint8_t> ciphertext;
+
+// Encrypt plaintext messages in 1K blocks.
+// NOTE: If `update()`-s and the following `read()` operations use the same chunk sizes you can
+// benefit from optimized zero-copy operations.
+const size_t packetSize = 1024;
+auto packetIterator = plaintext.begin();
+while (packetIterator < plaintext.end() - packetSize) {
+    std::vector<uint8_t> plaintextPacket{packetIterator, packetIterator + packetSize};
+    packetIterator += packetSize;
+    encryptor->update(plaintextPacket);
+    auto encryptedPacket = encryptor->read(packetSize);
+    std::copy(std::begin(encryptedPacket), std::end(encryptedPacket), std::back_inserter(ciphertext));
+}
+
+// AES GCM does not use padding and `finalEncryptedChunk` would be empty but you still
+// have to call `finish()` to indicate to the cipher that encryption is complete and
+// authentication tag can be computed.
+// Please also note that if you use other cipher in stream mode, `finish()` might actually
+// return some encrypted data.
+auto finalEncryptedChunk = encryptor->finish();
+std::copy(std::begin(finalEncryptedChunk), std::end(finalEncryptedChunk), std::back_inserter(ciphertext));
+
+auto iv = encryptor->getIV();
+auto authTag = encryptor->getAuthTag();
+
+//
+// Decryption
+//
+
+auto decryptor = AESCipherBuilder{SymmetricCipherMode::GCM, SymmetricCipherKeySize::S_256, secretKey}.setIV(iv).buildAuthenticatedDecryptor();
+
+
+// You can mix `update()`, `read()` and `readAll()`. For example, in the code below we first
+// read and decrypt several packets, then accumulated some decrypted blocks in the cipher
+// internal memory and finally reading all the remaining plaintext from the cipher.
+//
+// NOTE: `update()` and sequential `read()` of the same chunk size is still might be the most
```

Comment:
The same applies here, `readAll()` now also works here.

Reply:
Done